### PR TITLE
Make array value cookies able to be sent.

### PR DESCRIPTION
**Pull request by markstory**
Add a new method to get a cookie value as a string. I considered changing how `getValue()` works but thought that had bigger compatibility implications. Instead I've modified the interface/implementation. Normally I would shy away from changing interfaces, but this is a very new class and the likelihood of there be user-space implementations is quite low.

Refs #11208

### CODE REVIEW

File: src/Http/Cookie/CookieInterface.php [hunk 1]
```diff
@@ -49,6 +49,15 @@ public function getName();
     public function getValue();
 
     /**
+     * Gets the cookie value as a string.
+     *
+     * This will collapse any complex data in the cookie with json_encode()
+     *
+     * @return string
+     */
+    public function getStringValue();
```
Comment (jeremyharris) [inline src/Http/Cookie/CookieInterface.php]: Any reason not to use `__toString`?

File: src/Http/Cookie/Cookie.php [hunk 2]
```diff
@@ -245,20 +236,27 @@ protected function validateName($name)
...
 
     /**
-     * Gets the cookie value
-     *
-     * @return string|array
+     * {@inheritDoc}
      */
     public function getValue()
     {
         return $this->value;
```
Comment (stefanozoffoli) [inline src/Http/Cookie/Cookie.php]: I applied your fix and it correctly save the cookie now.
However when I read the cookie value, it continues to be in json format.

Can be correct to modify here like this?
```php
return $this->_expand($this->value);
```

But also when using `ServerRequest->getCookie($cookie_name);` it returns the cookie value in json format and it doesn't call `Cookie->getValue()`.

Comment (stefanozoffoli) [reply src/Http/Cookie/Cookie.php]: By setting
```php
return $this->_expand($this->value);
```
There are warnings in the `EncryptedCookieMiddleware` so it's more complicated than this :sweat_smile:

File: src/Http/Cookie/CookieInterface.php (hunk 1 above)
Comment (markstory) [reply src/Http/Cookie/CookieInterface.php]: Cookies have more properties than just the value. Wouldn't the string value of a cookie include the Secure, HttpOnly, Path and Domain bits?

File: src/Http/Cookie/Cookie.php (hunk 2 above)
Comment (markstory) [reply src/Http/Cookie/Cookie.php]: The value of the cookie is the JSON data though when you're reading from a request. The cookie middleware adds more behavior to cookies though. I wouldn't recommend people store JSON data in plaintext cookies as its full of 💣

File: src/Http/Cookie/CookieInterface.php (hunk 1 above)
Comment (jeremyharris) [reply src/Http/Cookie/CookieInterface.php]: Yep, you're absolutely right! Disregard :)

**Pull request merged**